compiler/src/scala/quoted/runtime/impl/QuoteMatcher.scala

```diff
@@ -244,12 +244,10 @@ object QuoteMatcher {
                 case Select(qual2, _) if symbolMatch(scrutinee, pattern) =>
                   ref match
                     case Select(qual1, _) => qual1 =?= qual2
-                    case ref: Ident =>
-                      ref.tpe match
-                        case TermRef(qual: TermRef, _) => tpd.ref(qual) =?= qual2
-                        case TermRef(qual: ThisType, _) if qual.classSymbol.is(Module, butNot = Package) =>
-                          tpd.ref(qual.classSymbol.companionModule) =?= qual2
-                        case _ => matched
+                    case ref: Ident if qual2.symbol == defn.QuotedRuntimePatterns_patternHole =>
```

Comment:
Why is patternHole special-cased now?

Reply:
If the pattern does not contain a hole, then we do not need to extract a value and can just call `matched` directly. The code was updated to cover holes in any part of the prefix.